### PR TITLE
Post and pre hooks for the mutation step

### PR DESCRIPTION
**Pull request by boxed**
This is a prototype of solving the problem in #84

### CODE REVIEW

File: mutmut/__main__.py [hunk 1]
```diff
@@ -506,6 +523,9 @@ def run_mutation(config, filename, mutation_id):
     if cached_status != UNTESTED:
         return cached_status
 
+    if config.pre_mutation:
+        subprocess.check_output(config.pre_mutation, shell=True)
```
Comment (sobolevn) [inline mutmut/__main__.py]: What will happen if any of the pre/post mutation process fails?

Comment (boxed) [reply mutmut/__main__.py]: Then it terminates mutmut. I think that makes sense as a default, otherwise it's pretty hard to debug the post/pre hooks.

Comment (sobolevn) [reply mutmut/__main__.py]: I guess the exception log from failing processes must be shown to the end user.

Comment (boxed) [reply mutmut/__main__.py]: Yea, that's what you get by default. In fact you'll get all the stdout and stderr of the called process so your pre/post hooks should probably be silent in the normal case to not destroy the output of mutmut.

Comment (sobolevn) [reply mutmut/__main__.py]: Awesome! 👍

File: mutmut/__main__.py [hunk 2]
```diff
@@ -533,6 +553,9 @@ def run_mutation(config, filename, mutation_id):
     finally:
         move(filename + '.bak', filename)
 
+        if config.post_mutation:
```
Comment (nklapste) [inline mutmut/__main__.py]: we should add a simple integration test that ads some coverage over these statements just for good measure.

Maybe something like:

```python
def test_post_mutation_hook(filesystem):
    result = CliRunner().invoke(climain, ['run', '--paths-to-mutate=foo.py', "--test-time-base=15.0", """--post-mutation='python -c "print(\'post mutation stub\')""""], catch_exceptions=False)
    print(repr(result.output))
    assert result.exit_code == 0
    assert "post mutation stub" in result.output
```

Comment (boxed) [reply mutmut/__main__.py]: Good idea. I modified the test a little bit from what you wrote, and it did show a bug! I swallowed the output which would have been super annoying.

File: tests/test_main.py [hunk 3]
```diff
@@ -339,3 +339,19 @@ def test_use_coverage(capsys, filesystem):
...
+def test_pre_and_post_mutation_hook(filesystem):
+    result = CliRunner().invoke(
+        climain, [
+            'run',
+            '--paths-to-mutate=foo.py',
+            "--test-time-base=15.0",
+            "--pre-mutation=echo pre mutation stub",
+            "--post-mutation=echo post mutation stub",
+        ], catch_exceptions=False)
+    print(result.output)
```
Comment (sobolevn) [inline tests/test_main.py]: Looks like this `print` is not required

Comment (boxed) [reply tests/test_main.py]: It isn't, but when the test fails it's super nice to have.

**Pull request merged**